crates/cheatcodes/src/script.rs

```diff
@@ -29,6 +32,68 @@ impl Cheatcode for broadcast_2Call {
     }
 }
 
+impl Cheatcode for attachDelegationCall {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { implementation, authority, v, r, s } = self;
+
+        let mut authority_acc =
+            ccx.ecx.journaled_state.load_account(*authority, &mut ccx.ecx.db)?;
+
+        let auth = Authorization {
+            address: *implementation,
+            nonce: authority_acc.data.info.nonce,
+            chain_id: ccx.ecx.env.cfg.chain_id,
+        };
+        let signed_auth = SignedAuthorization::new_unchecked(
+            auth,
+            *v,
+            U256::from_be_bytes(r.0),
+            U256::from_be_bytes(s.0),
+        );
+
+        // verify signature is from claimed authority
+        let recovered = signed_auth.recover_authority().map_err(|e| format!("{e}"))?;
+        if recovered != *authority {
+            return Err("invalid signature".into());
+        }
+
+        // write delegation code
+        let bytecode = Bytecode::new_eip7702(*implementation);
+        authority_acc.info.code = Some(bytecode.clone());
+        authority_acc.info.code_hash = bytecode.hash_slow();
+        authority_acc.mark_touch();
```

Comment:
there's `set_code` helper on `journaled_state` we can use, check out the `etch` implementation:
https://github.com/foundry-rs/foundry/blob/c4a31a624874ab36284fca4e48d2197e43a62fbe/crates/cheatcodes/src/evm.rs#L442-L451

Reply:
also it looks like we don't handle nonce here at all? we should account for delegation increasing the nonce

Reply:
Fixed both, thanks